docs/agents/sematext-agent/custom-logs.md

```diff
@@ -0,0 +1,179 @@
+  files: ["/path/to/myapp.log"]                                                     # log file paths
+  token: "your-sematext-logs-app-token"                                             # Sematext Cloud Logs App token
+  isJSON: false                                                                     # for JSON formatted files 
+  regexPatterns:                                                                    # regex patterns to parse logs
+    - '^(?P<timestamp>\d{4}-\d{2}-\d{2}[\s|T][\d+|:]+.\d+)\s(?P<message>[\s|\S]+)'
+  patternTypes:                                                                     # data types for each regex pattern
+    timestamp: "string"
+    message: "string"
+```
+
+## Configuration Details
+
+Each section in the `custom-logs.yml` file defines a log shipping rule. There are also several variables to configure how the Sematext Agent collects and parses your log files.
+
+### Log type
+
+- **Description**: The type of your logs. You can define your own type or use a pre-defined type.
+- **Required**: Yes
+- **Example**: `myapp-logs` which is a custom type, `nginx` which is pre-defined
+
+The log type can be a custom type defined by the user or a pre-defined type that maps to one of Sematext's supported Logs Apps (e.g., `nginx` for Nginx Logs). All available types can be found below.
+
+| Integration                          | Type                     |
+|--------------------------------------|--------------------------|
+| ActiveMQ                             | activemq                 |
+| Apache                               | apache                   |
+| Elasticsearch                        | elasticsearch            |
+| OpenSearch                           | opensearch               |
+| JVM apps                             | jvm                      |
+| Linux daemons                        | linux-daemon             |
+| MySQL                                | mysql                    |
+| Nginx                                | nginx                    |
+| Postgres                             | postgresql               |
+| RabbitMQ                             | rabbitmq                 |
+| Solr                                 | solr                     |
+| Varnish Cache                        | varnishcache             |
+
+### `files`
+
+- **Description**: Specifies the paths to the log files. Glob patterns are not supported.
+- **Required**: Yes
+- **Example**:
+
+```yaml
+  files: ["/path/to/myapp.log"]
+```
+
+### `token`
+
+- **Description**: The token for your Sematext Cloud Logs App.
+- **Required**: Yes
+- **Example**:
+
+```yaml
+  token: "af526914-15fe-481e-b06c-e7e07daf793c"
+```
+
+### `isJSON`
+
+- **Description**: Set to `true` if your log files are JSON formatted. If `true`, `regexPatterns` and `patternTypes` are not needed.
+- **Required**: No
+- **Default**: `false`
+- **Example**:
+
+  ```yaml
+  isJSON: true
+  ```
+
+### `regexPatterns`
```

Comment:
So we want to say what happens if one specifies a valid regex?

Reply:
maybe https://regex101.com? cc @boratanrikulu

Reply:
could be this one https://rustexp.lpil.uk/